--- a/drivers/media/video/mt9t112.c
+++ b/drivers/media/video/mt9t112.c
@@ -875,6 +875,12 @@ static int mt9t112_s_stream(struct v4l2_
 
 	ECHECKER(ret, mt9t112_init_camera(client));
 
+	/* for 30Hz */
+	mt9t112_mcu_write(ret, client, VAR(18, 12), 0x0024);
+	mt9t112_mcu_write(ret, client, VAR(26, 21), 0x0006);
+	mt9t112_mcu_write(ret, client, VAR(26, 23), 0x0007);
+	mt9t112_mcu_write(ret, client, VAR(26, 45), 0x0000);
+
 	/* Invert PCLK (Data sampled on falling edge of pixclk) */
 	mt9t112_reg_write(ret, client, 0x3C20,
 			  (MT9T112_FLAG_PCLK_RISING_EDGE & priv->info->flags) ?
