---
diff --git a/drivers/media/video/mt9t112.c b/drivers/media/video/mt9t112.c index bd5ef62..cd474be 100644
--- a/drivers/media/video/mt9t112.c
+++ b/drivers/media/video/mt9t112.c
@@ -880,8 +880,6 @@ static int mt9t112_s_stream(struct v4l2_
 			  (MT9T112_FLAG_PCLK_RISING_EDGE & priv->info->flags) ?
 			  0x0001 : 0x0000);
 
-	mdelay(100);
-
 	mt9t112_mcu_write(ret, client, VAR(26, 7), priv->format->fmt);
 	mt9t112_mcu_write(ret, client, VAR(26, 9), priv->format->order);
 	mt9t112_mcu_write(ret, client, VAR8(1, 0), 0x06);
@@ -899,6 +897,8 @@ static int mt9t112_s_stream(struct v4l2_
 
 	CLOCK_INFO(client, EXT_CLOCK);
 
+	mdelay(550);
+
 	return ret;
 }

--
1.6.3.3

