diff --git a/gfxdrivers/sh772x/kernel-module/sh7722.c b/gfxdrivers/sh772x/kernel-module/sh7722.c
index 981cf69..2d1accb 100644
--- a/gfxdrivers/sh772x/kernel-module/sh7722.c
+++ b/gfxdrivers/sh772x/kernel-module/sh7722.c
@@ -36,10 +36,6 @@
 
 /**********************************************************************************************************************/
 
-#ifndef SH7722_BEU_IRQ
-#define SH7722_BEU_IRQ 53
-#endif
-
 #ifndef SH7722_VEU_IRQ
 #define SH7722_VEU_IRQ 54
 #endif
@@ -767,18 +763,6 @@ sh7722_veu_irq( int irq, void *ctx )
 /**********************************************************************************************************************/
 
 static irqreturn_t
-sh7722_beu_irq( int irq, void *ctx )
-{
-     BEVTR = 0;
-
-     /* Nothing here so far. But Vsync could be added. */
-
-     return IRQ_HANDLED;
-}
-
-/**********************************************************************************************************************/
-
-static irqreturn_t
 sh7722_tdg_irq( int irq, void *ctx )
 {
      SH772xGfxSharedArea *shared = ctx;
@@ -1074,14 +1058,6 @@ sh7722_init( void )
              jpeg_order, jpeg_area, virt_to_phys(jpeg_area), SH7722GFX_JPEG_SIZE );
 
 
-     /* Register the BEU interrupt handler. */
-     ret = request_irq( SH7722_BEU_IRQ, sh7722_beu_irq, IRQF_DISABLED, "BEU", (void*) shared );
-     if (ret) {
-          printk( KERN_ERR "%s: request_irq() for interrupt %d failed! (error %d)\n",
-                  __FUNCTION__, SH7722_BEU_IRQ, ret );
-          goto error_beu;
-     }
-
 #ifdef SH7722GFX_IRQ_POLLER
      kernel_thread( sh7722_tdg_irq_poller, (void*) shared, CLONE_KERNEL );
 #else
@@ -1126,9 +1102,6 @@ error_jpu:
 
 error_tdg:
 #endif
-     free_irq( SH7722_BEU_IRQ, (void*) shared );
-
-error_beu:
      for (i=0; i<1<<jpeg_order; i++)
           ClearPageReserved( jpeg_page + i );
 
@@ -1168,8 +1141,6 @@ sh7722_exit( void )
      free_irq( SH7722_TDG_IRQ, (void*) shared );
 #endif
 
-     free_irq( SH7722_BEU_IRQ, (void*) shared );
-
      misc_deregister( &sh7722gfx_miscdev );
 
 
@@ -1190,3 +1161,4 @@ sh7722_exit( void )
 #endif
 }
 
+
diff --git a/gfxdrivers/sh772x/kernel-module/sh7723.c b/gfxdrivers/sh772x/kernel-module/sh7723.c
index ea64cf4..96b1c5d 100644
--- a/gfxdrivers/sh772x/kernel-module/sh7723.c
+++ b/gfxdrivers/sh772x/kernel-module/sh7723.c
@@ -39,10 +39,6 @@
 
 /**********************************************************************************************************************/
 
-#ifndef SH7723_BEU_IRQ
-#define SH7723_BEU_IRQ 53
-#endif
-
 #ifndef SH7723_TDG_IRQ
 #define SH7723_TDG_IRQ 44
 #endif
@@ -235,16 +231,6 @@ sh7723_wait_next( SH772xGfxSharedArea *shared )
 /**********************************************************************************************************************/
 
 static irqreturn_t
-sh7723_beu_irq( int irq, void *ctx )
-{
-     BEVTR = 0;
-
-     /* Nothing here so far. But Vsync could be added. */
-
-     return IRQ_HANDLED;
-}
-
-static irqreturn_t
 sh7723_tdg_irq( int irq, void *ctx )
 {
      SH772xGfxSharedArea *shared = ctx;
@@ -489,14 +475,6 @@ sh7723_init( void )
      printk( KERN_INFO "sh7723gfx: shared area (order %d) at %p [%lx] using %d bytes\n",
              shared_order, shared, virt_to_phys(shared), sizeof(SH772xGfxSharedArea) );
 
-     /* Register the BEU interrupt handler. */
-     ret = request_irq( SH7723_BEU_IRQ, sh7723_beu_irq, IRQF_DISABLED, "BEU", (void*) shared );
-     if (ret) {
-          printk( KERN_ERR "%s: request_irq() for BEU interrupt %d failed! (error %d)\n",
-                  __FUNCTION__, SH7723_BEU_IRQ, ret );
-          goto error_beu;
-     }
-
 #ifdef SH7723GFX_IRQ_POLLER
      kernel_thread( sh7723_tdg_irq_poller, (void*) shared, CLONE_KERNEL );
 #else
@@ -515,9 +493,6 @@ sh7723_init( void )
 
 
 error_tdg:
-     free_irq( SH7723_BEU_IRQ, (void*) shared );
-
-error_beu:
 #ifndef SHARED_AREA_PHYS
      for (i=0; i<1<<shared_order; i++)
           ClearPageReserved( shared_page + i );
@@ -551,8 +526,6 @@ sh7723_exit( void )
      free_irq( SH7723_TDG_IRQ, (void*) shared );
 #endif
 
-     free_irq( SH7723_BEU_IRQ, (void*) shared );
-
      misc_deregister( &sh7723gfx_miscdev );
 
 
@@ -564,3 +537,4 @@ sh7723_exit( void )
 #endif
 }
 
+
