diff -ruNp a/gfxdrivers/sh772x/sh7722.c b/gfxdrivers/sh772x/sh7722.c
--- a/gfxdrivers/sh772x/sh7722.c	2009-11-23 14:56:07.000000000 +0000
+++ b/gfxdrivers/sh772x/sh7722.c	2010-06-14 15:37:11.502287207 +0100
@@ -3,7 +3,6 @@
 #endif
 
 #include <stdio.h>
-#include <jpeglib.h>
 
 #undef HAVE_STDLIB_H
 
@@ -41,7 +40,6 @@ DFB_GRAPHICS_DRIVER( sh7722 )
 
 #include "sh7722.h"
 #include "sh7722_blt.h"
-#include "sh7722_jpeglib.h"
 #include "sh7722_layer.h"
 #include "sh7722_lcd.h"
 #include "sh7722_multi.h"
