diff --git a/configure.ac b/configure.ac
index aff5aea..9d5bab2 100644
--- a/configure.ac
+++ b/configure.ac
@@ -43,9 +43,10 @@ AC_CHECK_FUNCS([memset])
 # Check for Bellagio
 PKG_CHECK_MODULES([OMXIL], [libomxil-bellagio])
 
-_PKG_CONFIG([pluginsdir], [variable=pluginsdir], [libomxil-bellagio])
-BELLAGIO_PLUGINSDIR=${pkg_cv_pluginsdir}
-AC_SUBST([BELLAGIO_PLUGINSDIR])
+#_PKG_CONFIG([pluginsdir], [variable=pluginsdir], [libomxil-bellagio])
+#BELLAGIO_PLUGINSDIR=${pkg_cv_pluginsdir}
+BELLAGIO_PLUGINSDIR=/usr/lib/bellagio/
+AC_SUBST(BELLAGIO_PLUGINSDIR)
 
 # Bellagio debug levels: <bellagio/omx_comp_debug_levels.h>
 AC_ARG_ENABLE(

