--- qt-everywhere-opensource-src-4.7.0/src/corelib/tools/qregexp.cpp	2010-09-10 10:05:25.000000000 +0100
+++ qt-everywhere-opensource-src-4.7.0/src/corelib/tools/qregexp.cpp	2011-01-26 09:35:12.438653300 +0000
@@ -4240,11 +4240,13 @@
 
         for (int i = 0; i < n; i += 2) {
             QString m;
-            if (captured[i + 1] == 0)
+            if (captured[i + 1] == 0) {
                 m = QLatin1String(""); // ### Qt 5: don't distinguish between null and empty
-            else if (captured[i] >= 0)
+                priv->capturedCache.append(m);
+            } else if (captured[i] >= 0) {
                 m = priv->t.mid(captured[i], captured[i + 1]);
-            priv->capturedCache.append(m);
+                priv->capturedCache.append(m);
+            }
         }
         priv->t.clear();
     }
